pkg/distributor/distributor.go

```diff
@@ -560,14 +562,15 @@ func (d *Distributor) Push(ctx context.Context, req *logproto.PushRequest) (*log
 				discardedBytes := util.EntriesTotalSize(stream.Entries)
 				d.validator.reportDiscardedData(reason, validationContext, retentionHours, policy, discardedBytes, len(stream.Entries))
 
-				// If the status code is 200, return success.
+				// If the status code is 200, return no error.
 				// Note that we still log the error and increment the metrics.
 				if statusCode == http.StatusOK {
-					// do not add error to validationErrors.
 					continue
 				}
 
-				validationErrors.Add(err)
+				// return an error but do not add it to validationErrors
+				// otherwise client will get a 400 and will log it.
+				secondTierErr = httpgrpc.Errorf(statusCode, "%s", err.Error())
```

Comment:
I think this secondTierErr approach is a bit difficult to reason about and adds some noise to the already big function. But I don't see a considerably simpler way to do this.

What about naming this `successStatusOverride` instead? I think it reads a bit better.

Reply:
that's fair. let me apply it.